woocommerce/woocommerce
# Working add shipping class option and modal

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Adding "create new shipping class" option in Shipping class dropdown on product block editor, as well as modal.

Closes #37886 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout branch and enable `product-block-editor` feature flag with the WCA Test Helper plugin.
2. Navigate to Products -> Add new, and you should see the product block editor.
3. Navigate to Shipping tab, and select "Follow class" under "Shipping fee" heading.
4. This should reveal a drop-down menu with a list of any shipping classes.
5. Click "Add new shipping class" within the drop-down, which should reveal the shipping class modal.
6. Fill out the modal form with a shipping class and hit "Add"
7. This should add the shipping class and select it within the drop-down.
8. Save the product, and refresh to ensure the value persists.
